src/dev/build/tasks/package_json/find_used_dependencies.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import globby from 'globby';
+// @ts-ignore
+import { parseEntries, dependenciesParseStrategy } from '@kbn/babel-code-parser';
+
+async function getDependencies(cwd: string, entries: string[]) {
+  // Return the dependencies retrieve from the
+  // provided code entries (sanitized) and
+  // parseStrategy (dependencies one)
+  return Object.keys(await parseEntries(cwd, entries, dependenciesParseStrategy, {}));
+}
+
+export async function findUsedDependencies(listedPkgDependencies: any, baseDir: any, isOss: any) {
+  // Define the entry points for the server code in order to
+  // start here later looking for the server side dependencies
+  const mainCodeEntries = [
+    `${baseDir}/src/cli/dist.js`,
+    `${baseDir}/src/cli_keystore/dist.js`,
+    `${baseDir}/src/cli_plugin/dist.js`,
+  ];
+
+  const discoveredPluginEntries = await globby([
+    `${baseDir}/src/plugins/*/server/index.js`,
+    `!${baseDir}/src/plugins/**/public`,
+  ]);
+
+  // It will include entries that cannot be discovered
+  // statically as they are required with dynamic imports.
+  // In vis_type_timelion the problem is the loadFunctions()
+  // which dynamically requires all the files inside of function folders
+  // Another way would be to include an index file and import all the functions
+  // using named imports
+  const dynamicRequiredEntries = await globby([
+    `${baseDir}/src/plugins/vis_type_timelion/server/**/*.js`,
+  ]);
+
+  // Compose all the needed entries
+  const serverEntries = [...mainCodeEntries, ...discoveredPluginEntries, ...dynamicRequiredEntries];
+
+  if (!isOss) {
```

Comment:
We can get rid of `isOss` option and merge this with the `serverEntries` since these directories don't exist in the build output.

Reply:
Good suggestion, I will try it out!

Reply:
looks like it worked well